presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -62,6 +62,7 @@
     private final LinkedHashSet<VariableReferenceExpression> outputs;
     private final LinkedHashSet<VariableReferenceExpression> groupByColumns;
     private final LinkedHashMap<VariableReferenceExpression, SortOrder> topNColumnOrderingMap;
+    private final LinkedHashMap<VariableReferenceExpression, Selection> topNColumnSelections;
```

Comment:
I understand the fix, but I wanted to discuss a counter proposal: What if `selections` changes from being the list of final projections to simply being a map that maps a variable to its Pinot selection. That is, it contains all of the selections ever encountered. 

And we introduce another notion of a Final project list: List<VariableReference>. 

The problem that is happening is that `selections` is being overwritten and it no longer contains the ordering variables. What if we always union new things into it instead of overwriting it ? 

I am not sure whether this would pan out or perhaps regress something else. 

If you think my counter proposal is too much work, then just a note on naming: What do you think of foldinging topNColumnSelections and topNColumnOrderings into one ? Perhaps create a class OrderingInformation that contains (SortOrder, Selection) ? I find the name topNColumnSelections ambiguous and I wonder if fusing it with topNColumnOrderingMap would elide that concern.

Reply:
Both seem better ideas to me then having two maps.

Since we have not encountered any other case like this, in favor of less invasion I would prefer going with merging topNColumnSelections and topNColumnOrderings. Making the change